pkg/scheduler/actions/enqueue/enqueue.go

```diff
@@ -142,3 +143,47 @@ func (enqueue *Action) getOverCommitFactor(ssn *framework.Session) float64 {
 
 	return factor
 }
+
+func isCapabilitySatisfied(ssn *framework.Session, targetJob *api.JobInfo) bool {
+	queueID := targetJob.Queue
+	queue := ssn.Queues[queueID]
+
+	// If no capability is set, always enqueue the job.
+	if len(queue.Queue.Spec.Capability) == 0 {
+		klog.V(4).Infof("Capability of queue <%s> was not set, allow job <%s/%s> to Enqueue.",
+			queue.Name, targetJob.Namespace, targetJob.Name)
+		return true
+	}
+
+	// If capability is set, calculate allocated resource of queue which the target belongs to,
+	// compare the sum of allocated and Job minRequest resource with the capability resource
+	allocatedResource := api.EmptyResource()
```

Comment:
I think this way is very inefficient as for every job, it is calculated once
suggest adding a new field in `queueAttr`

Reply:
Maybe two filed "inqueueSource" & "allocatedSource" is a good way, i need to implement it after sorting out of state transition and resource computational logic.